### DISCUSSION TITLE
Change VAE in config.txt

### DISCUSSION THREAD

**Discussion by LuosVleer** [lllyasviel/Fooocus#3018] [Q&A] [answered]
When I boot up Fooocus it always loads with the "default (model)" Vae,
and the Pony checkpoint I use seems to mess up with it, (requiring the sdxl_vae), so I thought of changing the default VAE in the config.txt
"default_vae": "sdxl_vae",

This does seem to work as it shows properly in the advanced tab, but I get an error saying the following:
raise ValueError(exp)
ValueError: [Errno 2] No such file or directory: 'J:\\Fooocus\\Fooocus\\models\\vae\\sdxl_vae'
which results in Fooocus not running.

So I am wondering what I am doing wrong.

So far, with a bit of trial and error, I have managed to change anything I wanted in the config, so this one has me a bit confuzzled.

any help is appreciated! thanks :)

**Comment by Rogala** [comment]
Put the VAE file in the `x:\Fooocus\models\vae\ folder.`
Then in the advanced settings, select it
ADVANCED tab - Developer Debug Mode checkbox - Debug tools tab - Down - VAE group and select your option.

**Reply by LuosVleer** [reply]
Thanks for the reply!

The VAE is already there, and I know I can select it in the advanced tab as I mentioned.
My goal is to have it select that one automatically trough the config.txt.

which --again-- works, as it shows the change in the advanced tab settings, but for some reason does give an error.

**Reply by bsdesigner** [reply]
![изображение_2024-05-27_154951992](https://github.com/lllyasviel/Fooocus/assets/36730003/068b77c5-a7f7-497e-ad9d-8dc8a29e80ef)

**Reply by Rogala** [reply, marked as answer]
You want to use it on a regular basis, then just in the 
`config.txt` the full name of the file with the extension.
`“default_vae”: “sdxlVAE_sdxlVAE.safetensors”`

**Reply by LuosVleer** [reply]
doh! wouldnt be the first time I forgot to add the extension.

thanks so much!
*hugs* all around!

hmm, can only pick one answer.
